WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostListViewModel.kt

```diff
@@ -80,6 +108,104 @@ class ReaderPostListViewModel @Inject constructor(
         newsTracker.trackNewsCardExtendedInfoRequested(READER, item.version)
     }
 
+    fun loadSubFilters() {
+        launch {
+            val filterList = ArrayList<SubfilterListItem>()
+
+            filterList.add(SectionTitle(UiStringRes(R.string.reader_filter_sites_title)))
+            filterList.add(
+                    SiteAll(
+                        label = UiStringRes(R.string.reader_filter_all_sites),
+                        onClickAction = ::onSubfilterClicked,
+                        isSelected = (getCurrentSubfilterValue() is SiteAll)
+                    )
+            )
+
+            // Filtering Discover out
+            val followedBlogs = ReaderBlogTable.getFollowedBlogs().let { blogList ->
+                blogList.filter { blog ->
+                    !(blog.url.startsWith("https://discover.wordpress.com"))
+                }
+            }
+
+            for (blog in followedBlogs) {
+                filterList.add(Site(
+                        label = if (blog.name.isNotEmpty()) UiStringText(blog.name) else UiStringRes(
+                            R.string.reader_untitled_post),
+                        onClickAction = ::onSubfilterClicked,
+                        blog = blog,
+                        isSelected = (getCurrentSubfilterValue() is Site) &&
+                                (getCurrentSubfilterValue() as Site).blog.name == blog.name
+                ))
+            }
+
+            filterList.add(Divider)
+
+            filterList.add(SectionTitle(UiStringRes(R.string.reader_filter_tags_title)))
+
+            val tags = ReaderTagTable.getFollowedTags()
+
+            for (tag in tags) {
+                filterList.add(Tag(
+                        label = UiStringText(tag.tagTitle),
+                        onClickAction = ::onSubfilterClicked,
+                        tag = tag,
+                        isSelected = (getCurrentSubfilterValue() is Tag) &&
+                                (getCurrentSubfilterValue() as Tag).tag.tagTitle == tag.tagTitle
+                ))
+            }
+
+            _subFilters.postValue(filterList)
+        }
+    }
+
+    private fun onSubfilterClicked(filter: SubfilterListItem) {
```

Comment:
I don't think the `when` is necessary since the other types are not clickable, right?

Reply:
I was guided by an attempt to be more explicit, but I agree simpler is better for now. Changed here fed0e186b3eb4f9be44740543af84242f01b2cb3

Reply:
👍 If we ever have more than 1 click action, I'd rather have more click handle methods